paymentsheet/src/main/java/com/stripe/android/link/domain/LinkProminenceFeatureProvider.kt

```diff
@@ -0,0 +1,57 @@
+package com.stripe.android.link.domain
+
+import com.stripe.android.core.Logger
+import com.stripe.android.core.utils.FeatureFlags
+import com.stripe.android.link.gate.LinkGate
+import com.stripe.android.paymentsheet.state.LinkState
+import javax.inject.Inject
+
+/**
+ * Provider interface for Link feature prominence settings and behaviors, handling
+ * the logic for determining when to show special Link UI elements or trigger automatic
+ * behaviors to enhance the Link payment experience.
+ */
+internal interface LinkProminenceFeatureProvider {
+
+    /**
+     * On FlowController, upon showing the payment method list, if the user has selects Link, this
+     * method will determine if the 2FA dialog should be shown eagerly and confirmed as selection upon successful
+     * authentication.
+     */
+    fun show2FADialogOnLinkSelectedInFlowController(linkState: LinkState): Boolean
+}
+
+internal class DefaultLinkProminenceFeatureProvider @Inject constructor(
+    private val linkGateFactory: LinkGate.Factory,
+    private val logger: Logger,
+) : LinkProminenceFeatureProvider {
+
+    override fun show2FADialogOnLinkSelectedInFlowController(
+        linkState: LinkState,
+    ): Boolean {
+        val linkConfiguration = linkState.configuration
+
+        if (!FeatureFlags.linkProminenceInFlowController.isEnabled) {
+            logger.debug("Prominence disabled: Client side feature flag is disabled")
+            return false
+        }
+
+        if (linkConfiguration.suppress2faModal == true) {
```

Comment:
We might want to introduce a separate field for FlowController specifically, but let's decide that at a later point.